Modules/Devices/test/Devices.Application.Tests/Tests/Extensions/IEnumerableExtensionsTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Backbone.Modules.Devices.Application.Extensions;
+using FluentAssertions;
+using Xunit;
+using static System.Double;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Extensions;
+public class IEnumerableExtensionsTests
+{
+    [Fact]
+    public void Null_Parameter_Throws_ArgumentNullException()
+    {
+        // Act
+        var acting = () => ((List<int>)null).Split(10); ;
+
+        // Assert
+        acting.Should().Throw<ArgumentNullException>();
+    }
+
+    [Fact]
+    public void Empty_List_Returns_Empty_List_Of_Lists()
+    {
+        // Arrange
+        var list = new List<int>().AsReadOnly();
+
+        // Act
+        var result = list.Split(10);
+
+        // Assert
+        result.Should().BeEmpty("The argument list was empty.");
+    }
+
+    [Fact]
+    public void Large_List_Returns_Lists_Of_Lists()
+    {
+        // Arrange
+        var list = new List<int>();
+        for (var i = 0; i < 192; i++)
```

Comment:
I always try to make my test data as small as possible. This reduces the complexity of the test.

In this case, this means that I would probably use a total size of 3 and a batch size of 2. Or, if you want to test that multiple lists are returned, 5 and 2.